### PR TITLE
Crystal Core Quest

### PR DESCRIPTION
**Pull request by sLackeR-hub**
Crystal Core Quest plus update to Frost Haven including fixing Robert/Adding David.  Also added Crystal Core portal to Landblock 9F2C.

### CODE REVIEW

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48717 Crystal.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     , (48717,  16,         32) /* ItemUseable - Remote */
+     , (48717,  81,          2) /* MaxGeneratedObjects */
+     , (48717,  82,          0) /* InitGeneratedObjects */  
+     , (48717,  93,    6292508) /* PhysicsState - Ethereal, ReportCollisions, IgnoreCollisions, Gravity, ReportCollisionsAsEnvironment, EdgeSlide */
+     , (48717,  95,          8) /* RadarBlipColor - Yellow */
+     , (48717, 119,          1) /* Active */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (48717,   1, True ) /* Stuck */
+     , (48717,  19, False) /* Attackable */;
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48717 Crystal.sql:19]: Missing AiImmobile and NpcLooksLikeObject

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48723 Crystal.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     , (48723,  16,         32) /* ItemUseable - Remote */
+     , (48723,  81,          2) /* MaxGeneratedObjects */
+     , (48723,  82,          0) /* InitGeneratedObjects */  
+     , (48723,  93,    6292508) /* PhysicsState - Ethereal, ReportCollisions, IgnoreCollisions, Gravity, ReportCollisionsAsEnvironment, EdgeSlide */
+     , (48723,  95,          8) /* RadarBlipColor - Yellow */
+     , (48723, 119,          1) /* Active */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (48723,   1, True ) /* Stuck */
+     , (48723,  19, False) /* Attackable */;
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48723 Crystal.sql:19]: Missing AiImmobile and NpcLooksLikeObject

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Elemental/48711 Scorching Flame.sql [hunk 3]
```diff
@@ -0,0 +1,98 @@
+DELETE FROM `weenie` WHERE `class_Id` = 48711;
+
+INSERT INTO `weenie` (`class_Id`, `class_Name`, `type`, `last_Modified`)
+VALUES (48711, 'ace48711-scorchingflame', 10, '2019-02-10 00:00:00') /* Creature */;
+
+INSERT INTO `weenie_properties_int` (`object_Id`, `type`, `value`)
+VALUES (48711,   1,         16) /* ItemType - Creature */
+     , (48711,   6,         -1) /* ItemsCapacity */
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Elemental/48711 Scorching Flame.sql:8]: Missing CreatureType property; should be CreatureType - Elemental

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql [hunk 4]
```diff
@@ -0,0 +1,121 @@
+DELETE FROM `weenie` WHERE `class_Id` = 48712;
+
+INSERT INTO `weenie` (`class_Id`, `class_Name`, `type`, `last_Modified`)
+VALUES (48712, 'ace48712-chargolem', 10, '2019-02-10 00:00:00') /* Creature */;
+
+INSERT INTO `weenie_properties_int` (`object_Id`, `type`, `value`)
+VALUES (48712,   1,         16) /* ItemType - Creature */
+     , (48712,   6,         -1) /* ItemsCapacity */
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql:8]: Missing CreatureType property; should be CreatureType - Golem

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48718 Wall of Ice.sql [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+     , (48718,   6,         -1) /* ItemsCapacity */
+     , (48718,   7,         -1) /* ContainersCapacity */
+     , (48718,  16,          1) /* ItemUseable - No */
+     , (48718,  93,       1032) /* PhysicsState - ReportCollisions, Gravity */
+     , (48718, 290,          1) /* HearLocalSignals */
+     , (48718, 291,         10) /* HearLocalSignalsRadius */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (48718,   1, True ) /* Stuck */
+     , (48718,  19, False) /* Attackable */;
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48718 Wall of Ice.sql:17]: Missing AiImmobile and NpcLooksLikeObject

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48724 Wall of Ice.sql [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+     , (48724,   6,         -1) /* ItemsCapacity */
+     , (48724,   7,         -1) /* ContainersCapacity */
+     , (48724,  16,          1) /* ItemUseable - No */
+     , (48724,  93,       1032) /* PhysicsState - ReportCollisions, Gravity */
+     , (48724, 290,          1) /* HearLocalSignals */
+     , (48724, 291,         10) /* HearLocalSignalsRadius */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (48724,   1, True ) /* Stuck */
+     , (48724,  19, False) /* Attackable */;
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48724 Wall of Ice.sql:17]: Missing AiImmobile and NpcLooksLikeObject

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/PressurePlate/71870 Pressure Plate.sql [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+     , (71870, 33, 0, 3, 0, 999, 0, 1640.91) /* LifeMagic           Specialized */;
+     
+INSERT INTO `weenie_properties_emote` (`object_Id`, `category`, `probability`, `weenie_Class_Id`, `style`, `substyle`, `quest`, `vendor_Type`, `min_Health`, `max_Health`)
+VALUES (71870, 8 /* Activation */, 1, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
+
+SET @parent_id = LAST_INSERT_ID();
+
+INSERT INTO `weenie_properties_emote_action` (`emote_Id`, `order`, `type`, `delay`, `extent`, `motion`, `message`, `test_String`, `min`, `max`, `min_64`, `max_64`, `min_Dbl`, `max_Dbl`, `stat`, `display`, `amount`, `amount_64`, `hero_X_P_64`, `percent`, `spell_Id`, `wealth_Rating`, `treasure_Class`, `treasure_Type`, `p_Script`, `sound`, `destination_Type`, `weenie_Class_Id`, `stack_Size`, `palette`, `shade`, `try_To_Bond`, `obj_Cell_Id`, `origin_X`, `origin_Y`, `origin_Z`, `angles_W`, `angles_X`, `angles_Y`, `angles_Z`)
+VALUES (@parent_id, 0, 88 /* LocalSignal */, 0, 1, NULL, 'Scold', NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL)
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/PressurePlate/71870 Pressure Plate.sql:51]: Probably shouldn't have the LocalSignal send be the first action before the chat text, as the golem may disappear before the text is sent, which would look odd

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48713 Blazing Crystal.sql [hunk 8]
```diff
@@ -0,0 +1,107 @@
+DELETE FROM `weenie` WHERE `class_Id` = 48713;
+
+INSERT INTO `weenie` (`class_Id`, `class_Name`, `type`, `last_Modified`)
+VALUES (48713, 'ace48713-blazingcrystal', 10, '2019-02-10 00:00:00') /* Creature */;
+
+INSERT INTO `weenie_properties_int` (`object_Id`, `type`, `value`)
+VALUES (48713,   1,         16) /* ItemType - Creature */
+     , (48713,   6,         -1) /* ItemsCapacity */
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48713 Blazing Crystal.sql:8]: Missing CreatureType property; should be CreatureType - Crystal

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48715 Prism Shard.sql [hunk 9]
```diff
@@ -0,0 +1,95 @@
+DELETE FROM `weenie` WHERE `class_Id` = 48715;
+
+INSERT INTO `weenie` (`class_Id`, `class_Name`, `type`, `last_Modified`)
+VALUES (48715, 'ace48715-prismshard', 10, '2019-02-10 00:00:00') /* Creature */;
+
+INSERT INTO `weenie_properties_int` (`object_Id`, `type`, `value`)
+VALUES (48715,   1,         16) /* ItemType - Creature */
+     , (48715,   6,         -1) /* ItemsCapacity */
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48715 Prism Shard.sql:8]: Missing CreatureType property; should be CreatureType - Crystal

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48720 Crystal.sql [hunk 10]
```diff
@@ -0,0 +1,26 @@
+DELETE FROM `weenie` WHERE `class_Id` = 48720;
+
+INSERT INTO `weenie` (`class_Id`, `class_Name`, `type`, `last_Modified`)
+VALUES (48720, 'ace48720-crystal', 1, '2019-02-10 00:00:00') /* Generic */;
+
+INSERT INTO `weenie_properties_int` (`object_Id`, `type`, `value`)
+VALUES (48720,   1,        128) /* ItemType - Misc */
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48720 Crystal.sql:7]: Not a creature so should be in Generic/Misc subfolders

**Review by ziang4891 — CHANGES_REQUESTED** [review]

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48717 Crystal.sql (hunk 1 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48717 Crystal.sql:19]: Added.

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48723 Crystal.sql (hunk 2 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48723 Crystal.sql:19]: Added.

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48718 Wall of Ice.sql (hunk 5 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48718 Wall of Ice.sql:17]: Added

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48724 Wall of Ice.sql (hunk 6 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48724 Wall of Ice.sql:17]: Added

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/PressurePlate/71870 Pressure Plate.sql (hunk 7 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/PressurePlate/71870 Pressure Plate.sql:51]: With the delays in both it works out perfect.

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Elemental/48711 Scorching Flame.sql (hunk 3 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Elemental/48711 Scorching Flame.sql:9]: Added Fire Elemental type

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql (hunk 4 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql:8]: Added Golem Type

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48713 Blazing Crystal.sql (hunk 8 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48713 Blazing Crystal.sql:9]: Added Crystal Type

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48715 Prism Shard.sql (hunk 9 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48715 Prism Shard.sql:9]: Added Crystal Type

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48720 Crystal.sql (hunk 10 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Crystal/48720 Crystal.sql]: Moved.

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48718 Wall of Ice.sql [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+     , (48718,  16,          1) /* ItemUseable - No */
+     , (48718,  93,       1032) /* PhysicsState - ReportCollisions, Gravity */
+     , (48718, 290,          1) /* HearLocalSignals */
+     , (48718, 291,         10) /* HearLocalSignalsRadius */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (48718,   1, True ) /* Stuck */
+     , (48718,  19, False) /* Attackable */
+     , (48718,  52, True) /* Attackable */
+     , (48718,  83, True) /* Attackable */;
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48718 Wall of Ice.sql:19]: Comment for the above two properties isn't correct

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48724 Wall of Ice.sql [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+     , (48724,  16,          1) /* ItemUseable - No */
+     , (48724,  93,       1032) /* PhysicsState - ReportCollisions, Gravity */
+     , (48724, 290,          1) /* HearLocalSignals */
+     , (48724, 291,         10) /* HearLocalSignalsRadius */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (48724,   1, True ) /* Stuck */
+     , (48724,  19, False) /* Attackable */
+     , (48724,  52, True) /* Attackable */
+     , (48724,  83, True) /* Attackable */;
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48724 Wall of Ice.sql:19]: Comment for the above two properties isn't correct

**Review by ziang4891 — CHANGES_REQUESTED** [review]

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48718 Wall of Ice.sql (hunk 11 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48718 Wall of Ice.sql]: Fixed.

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48724 Wall of Ice.sql (hunk 12 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Wall/48724 Wall of Ice.sql]: Fixed

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+VALUES (48712,  2074,   2.15)  /* Gossamer Flesh */
+     , (48712,  2128,   2.18)  /* Ilservian's Flame */
+     , (48712,  2130,   2.15)  /* Infernae */
+     , (48712,  2170,   2.15)  /* Inferno's Gift */
+     , (48712,  2745,   2.15)  /* Flame Arc VII */;
+
+INSERT INTO `weenie_properties_create_list` (`object_Id`, `destination_Type`, `weenie_Class_Id`, `stack_Size`, `palette`, `shade`, `try_To_Bond`)
+VALUES (48712, 9,  2423,  1, 0, 0, False) /* Create Gem (2423) for ContainTreasure */
+     , (48712, 9,   273, 24, 0, 0, False) /* Create Pyreal (273) for ContainTreasure */
+     , (48712, 9,  2434,  0, 0, 0, False) /* Create Lesser Mana Stone (2434) for ContainTreasure */;
```
Comment (ziang4891) [inline Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql:122]: CreateList entries like this should be removed, as these entries are just representative of what the mob dropped when PCAP'ed.  To add random loot to mobs, it needs to be added via DeathTreasureDID that goes in the weenie_properties_d_i_d table.  An entry of **(48712,  35,        462) /* DeathTreasureType - Loot Tier: 3 */** would probably be appropriate

Comment (ziang4891) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql:122]: Also, golems like this one should have two heartbeat emotes for wiring up their sleep and wait animations, like https://github.com/ACEmulator/ACE-World-16PY-Patches/blob/173fd0605f8163efa5edf0d7d4dc1afeb65a5d71/Database/Patches/2005-07-ThroneOfDestiny/9%20WeenieDefaults/Creature/Golem/00200%20Mud%20Golem.sql#L135

**Review by ziang4891 — CHANGES_REQUESTED** [review]

**Review by ziang4891 — CHANGES_REQUESTED** [review]
init_Level values in weenie_properties_attribute_2nd table need to be fixed as described in content_recreation Discord channel

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Elemental/48711 Scorching Flame.sql [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+     , (48711, 24, 0, 3, 0,  50, 0, 0) /* Run                 Specialized */;
+
+INSERT INTO `weenie_properties_body_part` (`object_Id`, `key`, `d_Type`, `d_Val`, `d_Var`, `base_Armor`, `armor_Vs_Slash`, `armor_Vs_Pierce`, `armor_Vs_Bludgeon`, `armor_Vs_Cold`, `armor_Vs_Fire`, `armor_Vs_Acid`, `armor_Vs_Electric`, `armor_Vs_Nether`, `b_h`, `h_l_f`, `m_l_f`, `l_l_f`, `h_r_f`, `m_r_f`, `l_r_f`, `h_l_b`, `m_l_b`, `l_l_b`, `h_r_b`, `m_r_b`, `l_r_b`)
+VALUES (48711,  0, 32, 50,  0.5,   140,   100,   200,   100,   200,   200,  200,   200,    0, 1,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2) /* Head */
+     , (48711, 16, 32,  0,    0,   140,   100,   200,   100,   200,   200,  200,   200,    0, 2,  0.4,  0.4,  0.4,  0.4,  0.4,  0.4,  0.4,  0.4,  0.4,  0.4,  0.4,  0.4) /* Torso */
+     , (48711, 17, 32,  5, 0.75,   120,   100,   200,   100,   200,   200,  200,   200,    0, 2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2) /* Tail */
+     , (48711, 21, 32,  0,    0,   110,   40,   34,   34,   34,   34,   80,   34,    0, 2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2,  0.2) /* Wings */;     
+
```
[the remark this answers is not in the document]
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Elemental/48711 Scorching Flame.sql]: Fixed.

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql (hunk 13 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Golem/48712 Char Golem.sql:122]: Fixed.

File: Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Elemental/48711 Scorching Flame.sql (hunk 14 above)
[the remark this answers is not in the document]
Comment (sLackeR-hub) [reply Database/Patches/2012-12-AGrowingTwilight/9 WeenieDefaults/Creature/Elemental/48711 Scorching Flame.sql]: Fixed.

**Review by ziang4891 — DISMISSED** [review]

**Review by DridsGhost — APPROVED** [review]

File: Database/Patches/2012-12-AGrowingTwilight/8 QuestDefDB/blazingshardpickup.sql [hunk 15]
```diff
@@ -0,0 +1,4 @@
+DELETE FROM quest WHERE name = 'blazingshardpickup';
```
Comment (DridsGhost) [inline Database/Patches/2012-12-AGrowingTwilight/8 QuestDefDB/blazingshardpickup.sql:1]: Missing contracts, probably need to rename the quest entries.

David 
BuriedAliveStarted1212 
BuriedAliveCompleted1212

File: Database/Patches/2012-12-AGrowingTwilight/8 QuestDefDB/crystalcorestart.sql [hunk 16]
```diff
@@ -0,0 +1,4 @@
+DELETE FROM quest WHERE name = 'crystalcorestart';
```
Comment (DridsGhost) [inline Database/Patches/2012-12-AGrowingTwilight/8 QuestDefDB/crystalcorestart.sql:1]: Missing contracts, probably need to rename the quest entries.

David
BuriedAliveStarted1212
BuriedAliveCompleted1212

**Review by DridsGhost — CHANGES_REQUESTED** [review]
Missing contracts, probably need to rename the quest entries.

David
BuriedAliveStarted1212
BuriedAliveCompleted1212

**Review by DridsGhost — APPROVED** [review]
nice job

**Review by ziang4891 — APPROVED** [review]

**Pull request merged**